### ISSUE TITLE
Argument 1 passed to Facebook\WebDriver\Remote\JsonWireCompat::getElement() must be of the type array, string given

### ISSUE THREAD

**Issue by Mike-Benoit** [php-webdriver/php-webdriver#1062]
### Bug description

I'm seeing this type error exception only in some cases, but it seems outside of my control, here is the full trace:

```
Error: Argument 1 passed to Facebook\WebDriver\Remote\JsonWireCompat::getElement() must be of the type array, string given, called in ./vendor/php-webdriver/webdriver/lib/Remote/RemoteWebDriver.php on line 226
#0 ./vendor/php-webdriver/webdriver/lib/Remote/RemoteWebDriver.php(226): Facebook\WebDriver\Remote\JsonWireCompat::getElement()
#1 ./vendor/php-webdriver/webdriver/lib/WebDriverExpectedCondition.php(140): Facebook\WebDriver\Remote\RemoteWebDriver->findElement()
#2 [internal function]: Facebook\WebDriver\WebDriverExpectedCondition::Facebook\WebDriver\{closure}()
#3 ./vendor/php-webdriver/webdriver/lib/WebDriverWait.php(54): call_user_func()
#4 ./selenium.php(1870): Facebook\WebDriver\WebDriverWait->until()
#5 ./selenium.php(2496): Facebook\WebDriver\Selenium->navigateToHome()
```

### How could the issue be reproduced

Although I can't reproduce it consistently, it occurs about 60% of the time. The code is quite simple:

```php
$this->driver->wait( 60, 500 )->until( WebDriverExpectedCondition::presenceOfElementLocated( WebDriverBy::xpath( '//nav/ul/li/*[contains( text(), "Home")]' ) ) );
```

### Expected behavior

No PHP exception to be triggered.

### Php-webdriver version

1.14.0

### PHP version

7.4.3

### How do you start the browser driver or Selenium server

chromedriver

### Selenium server / Selenium Docker image version

4.8.1

### Browser driver (chromedriver/geckodriver...) version

111.0.5563.64

### Browser name and version

Chrome

### Operating system

Ubuntu 18.04

**Comment by Mike-Benoit** [comment]
I managed to get some additional information:

var_dump( WebDriverBy::xpath( '//nav/ul/li/*[contains( text(), "Home")]' ) )
```
object(Facebook\WebDriver\WebDriverBy)#40 (2) {
  ["mechanism":"Facebook\WebDriver\WebDriverBy":private]=>
  string(5) "xpath"
  ["value":"Facebook\WebDriver\WebDriverBy":private]=>
  string(40) "//nav/ul/li/*[contains( text(), "Home")]"
}
```

When findElement() returns a string that triggers this error, the string is:

```
/session/cc508b1529689c8f7b687c8cd8592217/element
```

**Comment by Mike-Benoit** [comment]
This seems to be occurring after the Selenium channel is expired. Its odd that we don't receive any exception from php-webdriver though. Here is the selenium log:

```
22:07:03.288 DEBUG [DefaultChannelPool$IdleChannelDetector.run] - Entry count for : http://localhost:8527 : 1
22:07:04.388 DEBUG [DefaultChannelPool$IdleChannelDetector.run] - Entry count for : http://localhost:8527 : 1
22:07:05.488 DEBUG [DefaultChannelPool$IdleChannelDetector.run] - Entry count for : http://localhost:8527 : 1
22:07:06.588 DEBUG [DefaultChannelPool$IdleChannelDetector.run] - Entry count for : http://localhost:8527 : 1
22:07:07.687 DEBUG [DefaultChannelPool$IdleChannelDetector.run] - Entry count for : http://localhost:8527 : 1
22:07:07.688 DEBUG [DefaultChannelPool$IdleChannelDetector.expiredChannels] - Adding Candidate expired Channel [id: 0x32e7518f, L:/127.0.0.1:47522 - R:localhost/127.0.0.1:8527] isIdleTimeoutExpired=true isRemotelyClosed=false isTtlExpired=false
22:07:07.694 DEBUG [DefaultChannelPool$IdleChannelDetector.closeChannels] - Closing Idle Channel [id: 0x32e7518f, L:/127.0.0.1:47522 - R:localhost/127.0.0.1:8527]
22:07:07.700 DEBUG [DefaultChannelPool$IdleChannelDetector.run] - Closed 1 connections out of 1 in 12 ms
22:07:07.707 FINEST [LoggingHandler.close] - [id: 0x32e7518f, L:/127.0.0.1:47522 - R:localhost/127.0.0.1:8527] CLOSE
22:07:07.768 FINEST [LoggingHandler.channelInactive] - [id: 0x32e7518f, L:/127.0.0.1:47522 ! R:localhost/127.0.0.1:8527] INACTIVE
22:07:07.774 DEBUG [AsyncHttpClientHandler.channelInactive] - Channel Closed: [id: 0x32e7518f, L:/127.0.0.1:47522 ! R:localhost/127.0.0.1:8527] with attribute DISCARD
22:07:07.776 FINEST [LoggingHandler.channelUnregistered] - [id: 0x32e7518f, L:/127.0.0.1:47522 ! R:localhost/127.0.0.1:8527] UNREGISTERED
22:07:40.139 DEBUG [LocalDistributor.lambda$asRunnableHealthCheck$9] - Running healthcheck for Node http://10.7.5.29:4444
22:07:40.141 DEBUG [LocalDistributor.updateNodeAvailability] - Health check result for http://10.7.5.29:4444 was UP
22:09:40.139 DEBUG [LocalDistributor.lambda$asRunnableHealthCheck$9] - Running healthcheck for Node http://10.7.5.29:4444
22:09:40.142 DEBUG [LocalDistributor.updateNodeAvailability] - Health check result for http://10.7.5.29:4444 was UP
22:11:06.801 DEBUG [RequestConverter.channelInactive] - Channel became inactive.
22:11:06.843 DEBUG [LoggingHandler.channelRead] - [id: 0x153fddcf, L:/0:0:0:0:0:0:0:0:4444] READ: [id: 0x262f5219, L:/0:0:0:0:0:0:0:1:4444 - R:/0:0:0:0:0:0:0:1:43052]
22:11:06.847 DEBUG [LoggingHandler.channelReadComplete] - [id: 0x153fddcf, L:/0:0:0:0:0:0:0:0:4444] READ COMPLETE
22:11:06.963 DEBUG [RequestConverter.channelRead0] - Incoming message: DefaultHttpRequest(decodeResult: success, version: HTTP/1.1)
POST /wd/hub/session/910d177d576480234a970b826ef8e833/frame HTTP/1.1
Host: localhost:4444
Content-Type: application/json;charset=UTF-8
Accept: application/json
Content-Length: 11
22:11:06.964 DEBUG [RequestConverter.channelRead0] - Start of http request: DefaultHttpRequest(decodeResult: success, version: HTTP/1.1)
POST /wd/hub/session/910d177d576480234a970b826ef8e833/frame HTTP/1.1
Host: localhost:4444
Content-Type: application/json;charset=UTF-8
Accept: application/json
Content-Length: 11
22:11:06.965 DEBUG [RequestConverter.channelRead0] - Incoming message: DefaultLastHttpContent(data: PooledSlicedByteBuf(ridx: 0, widx: 11, cap: 11/11, unwrapped: PooledUnsafeDirectByteBuf(ridx: 197, widx: 197, cap: 2048)), decoderResult: success)
22:11:06.965 DEBUG [RequestConverter.channelRead0] - End of http request: DefaultLastHttpContent(data: PooledSlicedByteBuf(ridx: 11, widx: 11, cap: 11/11, unwrapped: PooledUnsafeDirectByteBuf(ridx: 197, widx: 197, cap: 2048)), decoderResult: success)
22:11:06.974 DEBUG [HttpTracing.inject] - Injecting (POST) /session/910d177d576480234a970b826ef8e833/frame into OpenTelemetrySpan{traceId=457e7b62ba4c6d8f21510deed1e6e5da,spanId=db81f7fa35e0c7a6} at org.openqa.selenium.grid.router.HandleSession:121
22:11:06.992 DEBUG [SeleniumSpanExporter$1.lambda$export$4] - SpanData{spanContext=ImmutableSpanContext{traceId=457e7b62ba4c6d8f21510deed1e6e5da, spanId=d4621dcf174a6bd8, traceFlags=01, traceState=ArrayBasedTraceState{entries=[]}, remote=false, valid=true}, par
entSpanContext=ImmutableSpanContext{traceId=457e7b62ba4c6d8f21510deed1e6e5da, spanId=db81f7fa35e0c7a6, traceFlags=01, traceState=ArrayBasedTraceState{entries=[]}, remote=true, valid=true}, resource=Resource{schemaUrl=https://opentelemetry.io/schemas/1.16.0, att
ributes={service.name="unknown_service:java", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.22.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=default, version=null, schemaUrl=null, attributes={}}, name
=reverse_proxy, kind=INTERNAL, startEpochNanos=1680066666976000000, endEpochNanos=1680066666991532198, attributes=AttributesMap{data={http.status_code=404, http.target=/session/910d177d576480234a970b826ef8e833/frame, http.method=POST}, capacity=128, totalAddedV
alues=3}, totalAttributeCount=3, events=[], totalRecordedEvents=0, links=[], totalRecordedLinks=0, status=ImmutableStatusData{statusCode=ERROR, description=Kind: NOT_FOUND Description:}, hasEnded=true}
22:11:06.993 DEBUG [SeleniumSpanExporter$1.lambda$export$4] - SpanData{spanContext=ImmutableSpanContext{traceId=457e7b62ba4c6d8f21510deed1e6e5da, spanId=db81f7fa35e0c7a6, traceFlags=01, traceState=ArrayBasedTraceState{entries=[]}, remote=false, valid=true}, parentSpanContext=ImmutableSpanContext{traceId=00000000000000000000000000000000, spanId=0000000000000000, traceFlags=00, traceState=ArrayBasedTraceState{entries=[]}, remote=false, valid=false}, resource=Resource{schemaUrl=https://opentelemetry.io/schemas/1.16.0, attributes={service.name="unknown_service:java", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.22.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=default, version=null, schemaUrl=null, attributes={}}, name=router.handle_session, kind=INTERNAL, startEpochNanos=1680066666973000000, endEpochNanos=1680066666992551792, attributes=AttributesMap{data={http.status_code=404, http.target=/session/910d177d576480234a970b826ef8e833/frame, session.id=910d177d576480234a970b826ef8e833, http.method=POST}, capacity=128, totalAddedValues=4}, totalAttributeCount=4, events=[], totalRecordedEvents=0, links=[], totalRecordedLinks=0, status=ImmutableStatusData{statusCode=ERROR, description=Kind: NOT_FOUND Description:}, hasEnded=true}
22:11:07.002 DEBUG [RequestConverter.channelRead0] - Incoming message: DefaultHttpRequest(decodeResult: success, version: HTTP/1.1)
POST /wd/hub/session/910d177d576480234a970b826ef8e833/element HTTP/1.1
Host: localhost:4444
Content-Type: application/json;charset=UTF-8
Accept: application/json
Content-Length: 75
22:11:07.002 DEBUG [RequestConverter.channelRead0] - Start of http request: DefaultHttpRequest(decodeResult: success, version: HTTP/1.1)
POST /wd/hub/session/910d177d576480234a970b826ef8e833/element HTTP/1.1
Host: localhost:4444
Content-Type: application/json;charset=UTF-8
Accept: application/json
Content-Length: 75
22:11:07.002 DEBUG [RequestConverter.channelRead0] - Incoming message: DefaultLastHttpContent(data: PooledSlicedByteBuf(ridx: 0, widx: 75, cap: 75/75, unwrapped: PooledUnsafeDirectByteBuf(ridx: 263, widx: 263, cap: 2048)), decoderResult: success)
22:11:07.003 DEBUG [RequestConverter.channelRead0] - End of http request: DefaultLastHttpContent(data: PooledSlicedByteBuf(ridx: 75, widx: 75, cap: 75/75, unwrapped: PooledUnsafeDirectByteBuf(ridx: 263, widx: 263, cap: 2048)), decoderResult: success)
22:11:07.004 DEBUG [HttpTracing.inject] - Injecting (POST) /session/910d177d576480234a970b826ef8e833/element into OpenTelemetrySpan{traceId=d0edd4356bbeedebf4271bd30d0e2097,spanId=39e065f49f5a3bb8} at org.openqa.selenium.grid.router.HandleSession:121
22:11:07.007 DEBUG [SeleniumSpanExporter$1.lambda$export$4] - SpanData{spanContext=ImmutableSpanContext{traceId=d0edd4356bbeedebf4271bd30d0e2097, spanId=b672b5c37fb1261d, traceFlags=01, traceState=ArrayBasedTraceState{entries=[]}, remote=false, valid=true}, parentSpanContext=ImmutableSpanContext{traceId=d0edd4356bbeedebf4271bd30d0e2097, spanId=39e065f49f5a3bb8, traceFlags=01, traceState=ArrayBasedTraceState{entries=[]}, remote=true, valid=true}, resource=Resource{schemaUrl=https://opentelemetry.io/schemas/1.16.0, attributes={service.name="unknown_service:java", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.22.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=default, version=null, schemaUrl=null, attributes={}}, name=reverse_proxy, kind=INTERNAL, startEpochNanos=1680066667004000000, endEpochNanos=1680066667006924658, attributes=AttributesMap{data={http.target=/session/910d177d576480234a970b826ef8e833/element, http.method=POST}, capacity=128, totalAddedValues=2}, totalAttributeCount=2, events=[], totalRecordedEvents=0, links=[], totalRecordedLinks=0, status=ImmutableStatusData{statusCode=UNSET, description=}, hasEnded=true}
22:11:07.059 DEBUG [SeleniumSpanExporter$1.lambda$export$4] - SpanData{spanContext=ImmutableSpanContext{traceId=d0edd4356bbeedebf4271bd30d0e2097, spanId=39e065f49f5a3bb8, traceFlags=01, traceState=ArrayBasedTraceState{entries=[]}, remote=false, valid=true}, parentSpanContext=ImmutableSpanContext{traceId=00000000000000000000000000000000, spanId=0000000000000000, traceFlags=00, traceState=ArrayBasedTraceState{entries=[]}, remote=false, valid=false}, resource=Resource{schemaUrl=https://opentelemetry.io/schemas/1.16.0, attributes={service.name="unknown_service:java", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.22.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=default, version=null, schemaUrl=null, attributes={}}, name=router.handle_session, kind=INTERNAL, startEpochNanos=1680066667004000000, endEpochNanos=1680066667058359695, attributes=AttributesMap{data={error=true, http.target=/session/910d177d576480234a970b826ef8e833/element, session.id=910d177d576480234a970b826ef8e833, http.method=POST}, capacity=128, totalAddedValues=4}, totalAttributeCount=4, events=[ImmutableEventData{name=exception, attributes={exception.message="Unable to execute request for an existing session: Cannot find session with id: 910d177d576480234a970b826ef8e833
Build info: version: '4.8.1', revision: '8ebccac989'
System info: os.name: 'Linux', os.arch: 'amd64', os.version: '5.4.0-139-generic', java.version: '1.8.0_362'
Driver info: driver.version: unknown", exception.stacktrace="org.openqa.selenium.NoSuchSessionException: Cannot find session with id: 910d177d576480234a970b826ef8e833
Build info: version: '4.8.1', revision: '8ebccac989'
System info: os.name: 'Linux', os.arch: 'amd64', os.version: '5.4.0-139-generic', java.version: '1.8.0_362'
Driver info: driver.version: unknown
        at org.openqa.selenium.grid.node.local.LocalNode.executeWebDriverCommand(LocalNode.java:469)
        at org.openqa.selenium.remote.http.AddSeleniumUserAgent.lambda$apply$0(AddSeleniumUserAgent.java:42)
        at org.openqa.selenium.remote.http.Filter.lambda$andFinally$1(Filter.java:56)
        at org.openqa.selenium.remote.AddWebDriverSpecHeaders.lambda$apply$0(AddWebDriverSpecHeaders.java:35)
        at org.openqa.selenium.remote.http.Filter.lambda$andFinally$1(Filter.java:56)
        at org.openqa.selenium.grid.security.AddSecretFilter.lambda$apply$0(AddSecretFilter.java:40)
        at org.openqa.selenium.remote.http.Filter.lambda$andFinally$1(Filter.java:56)
        at org.openqa.selenium.grid.node.CustomLocatorHandler.execute(CustomLocatorHandler.java:134)
        at org.openqa.selenium.remote.http.Route$CombinedRoute.handle(Route.java:336)
        at org.openqa.selenium.remote.http.Route.execute(Route.java:68)
        at org.openqa.selenium.grid.node.Node.execute(Node.java:248)
        at org.openqa.selenium.grid.web.CombinedHandler.execute(CombinedHandler.java:59)
        at org.openqa.selenium.grid.web.RoutableHttpClientFactory$1.execute(RoutableHttpClientFactory.java:72)
        at org.openqa.selenium.grid.web.ReverseProxyHandler.execute(ReverseProxyHandler.java:92)
        at org.openqa.selenium.grid.router.HandleSession.execute(HandleSession.java:122)
        at org.openqa.selenium.remote.http.Route$PredicatedRoute.handle(Route.java:373)
        at org.openqa.selenium.remote.http.Route.execute(Route.java:68)
        at org.openqa.selenium.remote.http.Route$CombinedRoute.handle(Route.java:336)
        at org.openqa.selenium.remote.http.Route.execute(Route.java:68)
        at org.openqa.selenium.grid.router.Router.execute(Router.java:91)
        at org.openqa.selenium.grid.web.CheckOriginHeader.lambda$apply$0(CheckOriginHeader.java:66)
        at org.openqa.selenium.grid.web.CheckContentTypeHeader.lambda$apply$0(CheckContentTypeHeader.java:70)
        at org.openqa.selenium.grid.web.EnsureSpecCompliantResponseHeaders.lambda$apply$0(EnsureSpecCompliantResponseHeaders.java:34)
        at org.openqa.selenium.remote.http.Filter$1.execute(Filter.java:64)
        at org.openqa.selenium.remote.http.Route$CombinedRoute.handle(Route.java:336)
        at org.openqa.selenium.remote.http.Route.execute(Route.java:68)
        at org.openqa.selenium.remote.http.Route$NestedRoute.handle(Route.java:270)
        at org.openqa.selenium.remote.http.Route.execute(Route.java:68)
        at org.openqa.selenium.remote.http.Route$CombinedRoute.handle(Route.java:336)
        at org.openqa.selenium.remote.http.Route.execute(Route.java:68)
        at org.openqa.selenium.remote.http.Route$CombinedRoute.handle(Route.java:336)
        at org.openqa.selenium.remote.http.Route.execute(Route.java:68)
        at org.openqa.selenium.remote.http.Route$CombinedRoute.handle(Route.java:336)
        at org.openqa.selenium.remote.http.Route.execute(Route.java:68)
        at org.openqa.selenium.remote.http.Route$CombinedRoute.handle(Route.java:336)
        at org.openqa.selenium.remote.http.Route.execute(Route.java:68)
        at org.openqa.selenium.remote.AddWebDriverSpecHeaders.lambda$apply$0(AddWebDriverSpecHeaders.java:35)
        at org.openqa.selenium.remote.ErrorFilter.lambda$apply$0(ErrorFilter.java:44)
        at org.openqa.selenium.remote.http.Filter$1.execute(Filter.java:64)
        at org.openqa.selenium.remote.ErrorFilter.lambda$apply$0(ErrorFilter.java:44)
        at org.openqa.selenium.remote.http.Filter$1.execute(Filter.java:64)
        at org.openqa.selenium.netty.server.SeleniumHandler.lambda$channelRead0$0(SeleniumHandler.java:44)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:750)
", exception.type="org.openqa.selenium.NoSuchSessionException", http.flavor=1, http.handler_class="org.openqa.selenium.grid.router.HandleSession", http.host="localhost:4444", http.method="POST", http.request_content_length="75", http.scheme="HTTP", http.target="/session/910d177d576480234a970b826ef8e833/element", session.id="910d177d576480234a970b826ef8e833"}, epochNanos=1680066667057931474, totalAttributeCount=11}], totalRecordedEvents=1, links=[], totalRecordedLinks=0, status=ImmutableStatusData{statusCode=ERROR, description=Kind: CANCELLED Description:}, hasEnded=true}
22:11:07.062 WARN [SeleniumSpanExporter$1.lambda$export$1] - Unable to execute request for an existing session: Cannot find session with id: 910d177d576480234a970b826ef8e833
Build info: version: '4.8.1', revision: '8ebccac989'
System info: os.name: 'Linux', os.arch: 'amd64', os.version: '5.4.0-139-generic', java.version: '1.8.0_362'
Driver info: driver.version: unknown

```

**Comment by OndraM** [comment]
The error will be handled by exception (and will not cause an fatal error), it was implemented in #1064  and will be part of next release. 

Thanks for the report!
